src/groupByKey.ts

```diff
@@ -1,7 +1,37 @@
 type GroupsMap<T> = {
-  [key: string]: T[];
+  [key: string]: T[]; // Використовуємо string як тип для ключів
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  if (items.length === 0) {
+    return {};
+  }
+
+  const result: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const groupKey = item[key];
+
+    const groupKeyStringified = String(groupKey);
+
+    if (!result[groupKeyStringified]) {
+      result[groupKeyStringified] = [];
+    }
+
+    result[groupKeyStringified].push(item);
+  });
+
+  Object.keys(result).forEach((groupKey) => {
+    const groupItems = result[groupKey];
+    const stringSort = (a: T, b: T): number =>
+      String(a[key]).localeCompare(String(b[key]));
```

Comment:
The use of `localeCompare` is appropriate for sorting strings, but be aware that it may not behave as expected for non-English locales or special characters. Ensure this is the desired behavior for your application.